To-do list/Models/StatusButton.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  StatusButton.swift
+//  To-do list
+//
+//  Created by Kyrylo Tokar on 2022-09-16.
+//
+
+import UIKit
+
+class StatusButton {
+    
+    static func updateStatus(_ sender: UIButton, _ status: Task.Status) {
+        sender.setTitle(status.rawValue, for: .normal)
+        
+        switch status {
+        case .inProgress:
+            sender.backgroundColor = .systemOrange
+        case .done:
+            sender.backgroundColor = .systemRed
+        case .todo:
+            sender.backgroundColor = .systemGreen            
+        }
+    }
+    
+    static func changeStatus(_ status: Task.Status) -> Task.Status {
+        var changeStatus = status
+        
+        switch changeStatus {
+        case .todo:
+            changeStatus = .inProgress
+        case .inProgress:
+            changeStatus = .done
+        case .done:
+            changeStatus = .todo
+        }
+        return changeStatus
```

Comment:
Move this logic to Task.Status computed var:

`    
struct Task: Codable, Equatable {
     var title: String = ""
     var description: String = ""
     var deadline: Date = .now
     var status: Status = .todo
     var id: Int = 0

     enum Status: String, Codable {
         case todo = "to do"
         case inProgress = "in progress"
         case done = "done"
         
        var nextState: TaskModel.State {
              switch task.status {
              case .todo: return .inProgress
              case .inProgress: return .done
              case .done: return .todo
        }
     }
    }`

Reply:
moved